### PR TITLE
Type error in `fit` method from `lightning.pytorch.Trainer` when running training from Mac with mps

### LINKED ISSUE #612: [BUG] Type error in `fit` method from `lightning.pytorch.Trainer` when running training from Mac with mps
### 🐛 Bug Report

In the new version of `ligtning`, when initializing `Trainer`, the parameter `accelerator` is set to `"auto"` by default. On new versions of MacBook this means that `device="mps"`. However, training on mps must occur with float32. Currently, the library supplies data in float64. We need to fix this to take advantage of the Macbook's capabilities.

### Expected behavior

Training dl models runs on mps without errors.

Possible solution:
- Cast our data to float32 when training DL models. Maybe this can be done in the method `fit` from `DeepBaseModel` or in the method `to_torch_dataset` from `TSDataset`.
- Remove `self.trainer_params["accelerator"] = "cpu"` from method `raw_fit` in `DeepBaseModel`.
- Check that the tests pass and there are no errors.

### How To Reproduce

- Remove `self.trainer_params["accelerator"] = "cpu"` from method `raw_fit` in DeepBaseModel.
- Run any test with training DL model, for example 'test_mlp_model_run_weekly_overfit_with_scaler' from 'tests/test_models/test_nn/test_mlp.py'

### Environment

_No response_

### Additional context

_No response_

### Checklist

- [x] Bug appears at the latest library version

### PR DESCRIPTION
**Pull request by DanilSmorchkov**
## Before submitting (must do checklist)

- [x] Did you read the [contribution guide](https://github.com/tinkoff-ai/etna/blob/master/CONTRIBUTING.md)?
- [x] Did you update the docs? We use Numpy format for all the methods and classes.
- [x] Did you write any new necessary tests?
- [x] Did you update the [CHANGELOG](https://github.com/tinkoff-ai/etna/blob/master/CHANGELOG.md)?

## Proposed Changes
See #612

## Closing issues
Closed #612

### CODE REVIEW

File: tests/test_models/test_nn/test_patchtst.py [hunk 1]
```diff
@@ -22,7 +22,7 @@ def test_patchtst_model_run_weekly_overfit_with_scaler_small_patch(ts_dataset_we
     encoder_length = 14
     decoder_length = 14
     model = PatchTSTModel(
-        encoder_length=encoder_length, decoder_length=decoder_length, patch_len=1, trainer_params=dict(max_epochs=20)
+        encoder_length=encoder_length, decoder_length=decoder_length, patch_len=1, trainer_params=dict(max_epochs=22)
```
[the remark this answers is not in the document]
Comment (DanilSmorchkov) [reply tests/test_models/test_nn/test_patchtst.py:25]: When `max_epochs=20` the model on MPS does not converge, when `max_epoch=21` the model in ci does not converge. The difference in deviations is small.

File: etna/models/nn/mlp.py [hunk 2]
```diff
@@ -146,12 +146,12 @@ def make_samples(self, df: pd.DataFrame, encoder_length: int, decoder_length: in
         values_real = (
             df.drop(["target", "segment", "timestamp"] + list(self.embedding_sizes.keys()), axis=1)
             .select_dtypes(include=[np.number])
-            .values
+            .values.astype(np.float32)
```
[the remark this answers is not in the document]
Comment (DanilSmorchkov) [reply etna/models/nn/mlp.py:149]: To prevent casting to `float64` if, for example, we get an empty tensor or columns with `int64` / `int32`. There was no such problem in the tests, but I decided to add it just in case.

File: etna/models/base.py [hunk 3]
```diff
@@ -534,6 +535,11 @@ def __init__(
         self.split_params = {} if split_params is None else split_params
         self.trainer: Optional[Trainer] = None
 
+        if torch.mps.is_available() and self.trainer_params.get("accelerator", None) != "cpu":
+            warning_message = "If you use MPS sometimes it can cause unexpected results.\
```
Comment (d-a-bunin) [inline etna/models/base.py]: Better to write: 
```
If you use MPS, it can sometimes cause unexpected results.
If this happens try setting `accelerate="cpu"` in `trainer_params`.
```

File: tests/test_models/test_nn/test_mlp.py [hunk 4]
```diff
@@ -217,3 +218,20 @@ def test_params_to_tune(model, example_tsds):
     ts.fit_transform(transforms)
     assert len(model.params_to_tune()) > 0
     assert_sampling_is_valid(model=model, ts=ts)
+
+
+@patch("torch.mps.is_available", return_value=True)
```
Comment (d-a-bunin) [inline tests/test_models/test_nn/test_mlp.py:223]: We could rework it and put into `test_base probably. But I'm not sure about that.

Comment (DanilSmorchkov) [reply tests/test_models/test_nn/test_mlp.py:223]: How rework? I thought to put it in `test_base`. Is it ok to initialize the base class?

**Review by d-a-bunin — DISMISSED** [review]

**Review by d-a-bunin — APPROVED** [review]
